packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -225,30 +255,16 @@ abstract class Domain {
       }
       throw 'command not understood: $name.$command';
     }).then<dynamic>((dynamic result) {
-      if (result == null) {
-        _send(<String, dynamic>{'id': id});
-      } else {
-        _send(<String, dynamic>{'id': id, 'result': _toJsonable(result)});
-      }
-    }).catchError((dynamic error, dynamic trace) {
-      _send(<String, dynamic>{
-        'id': id,
-        'error': _toJsonable(error),
-        'trace': '$trace',
-      });
+      daemon.connection.sendResponse(id, _toJsonable(result));
+    }).catchError((Object error, StackTrace stackTrace) {
+      daemon.connection.sendErrorResponse(id, _toJsonable(error), stackTrace);
     });
   }
 
   void sendEvent(String name, [ dynamic args ]) {
-    final Map<String, dynamic> map = <String, dynamic>{'event': name};
-    if (args != null) {
-      map['params'] = _toJsonable(args);
-    }
-    _send(map);
+    daemon.connection.sendEvent(name, args != null ? _toJsonable(args) : null);
```

Comment:
`_toJsonable` returns `null` when `args` is null.

```suggestion
    daemon.connection.sendEvent(name, _toJsonable(args));
```